### ISSUE TITLE
URGENT. Full compatibility with WPML

### ISSUE THREAD

**Issue by TheBigBanik** [CrocoBlock/suggestions#1663]
Hi, since Jet Plugins are awesome, they are not suitable for multilanguage websites. So, if we need to create amazing websites, you should concider improving some compatibility things.

Make any kind of meta field created with jetengine translatable with WPML.
     1.1. Checkboxes. We can not translate the labels. For example the value in the database is fireplace and the label is fireplace. In wpml translation the value should stay fireplace in the database, no need to translate that but the value must be able to be translated in other languages. Which doesn't.
     1.2 Radio. Same as above
     1.3. Select. Same as above
In conclusion, the only translatable functions are the Jet Engine Custom Post types and taxonomies. Imagine a website like a car broker with many checkboxes and radio and select fields. It can not sell to other countries. It is super difficult to create a website like that.

I hope you will take under serious concideration all that missing compatible features so you can have the only one solution for websites. Thanks.

**Comment by xavipijuan** [comment]
+1 
Please Dev. Team, take this in consideration because it's really anoying to work with WPML and can't translate those fields directly.
I can add  to the list, the related post field. It have the same problem. if you have in the original version, several items related... when you came to translated version it's a crazy tryup to add those related posts in the translated version... (excuse me for my english)

**Comment by oceandiveloper** [comment]
+1
I'm just working on a WPML translation project for a client's website done with JetEngine and had to realize that JetEngine is not fully integrated with WPML yet. Multilanguage is so important and WPML one of the "big" plugins. Please make this work soon.

**Comment by jlcobb22** [comment]
+1
Any news on this?

**Comment by WardenCommander** [comment]
+1

**Comment by TheBigBanik** [comment]
A couple of weeks ago I contacted the WPML team and they offered a temporary solution that it is working for the checkbox fields. Unfortunately I am not in position right now to share it, I will try in the next 5-6 hours to paste the solution in a new reply so everyone will be able to work until a permanent fix.

**Comment by franc901** [comment]
+1

**Comment by TheBigBanik** [comment]
Hi everyone, I am giving you below the temporary solution from the WPML team. I tried it and it works. You will also have to add the code for radio and select fields to the relevant php files.
These are the steps:
1. Add the labels to WPML > String translations following these steps:
https://wpml.org/documentation/getting-started-guide/theme-localization/translating-options-coming-from-themes-and-plugins/
2. Translate the strings
3. In the user profile you need to enable "Set admin language as editing language.". If not, the strings will show in the language that the user selected in his profile.
4. Finally, the hacky part, in wp-content/plugins/jet-engine/framework/interface-builder/inc/controls/checkbox.php after line 89 add:
$option_label = __($option_label, 'admin_texts_jet_engine_meta_boxes');

I hope that will help you for now.

**Comment by acornavi** [comment]
Thanks for sharing the snippet to make WPML integration. It is not the way. WPML should work smoothly with the JetEngine and all other modules. It is hard to work with a half-baked plugin. Not reliable.
Please devs of Crocoblock. It is not a big deal to make full WPML compatibility. Once is done it will increase sales and usage of your wonderful plugin which is unfortunately not 100% useful in this stage of development. Thanks

**Comment by tony-keller** [comment]
Annoying to witness a complete lack of answer on such a topic.. I just landed a project thinking this was ready and am now in the unknown. @Crocoblock

**Comment by incircolo** [comment]
Me too, and now i have to re-design a lot of things to have translation available.

@TheBigBanik is this fix still working? I can't get it working unfortunately.

**Comment by TheBigBanik** [comment]
Hi guys, I am sorry to hear that there is still this incompatibility after more than a year I have mentioned it to both WPML and Crocoblock teams, so I contacted them again today. The WPML support team informed me that this is known issue to both teams and they are trying to solve it. On the other hand, sincce then many settings have been changed in the way that WPML sees the crocoblock meta fields so the workaround here is the following:
STEP 1: Create the metafields in the default language
STEP 2: WPML > Settings > Custom Fields Translation section and make those fields as translatable
STEP 3: WPML > String translation > Find those metafields and translate them to second language
STEP 4: Create a post in the default language and check the desired checkboxes (save post)
STEP 5: Press the + to translate and set those fields as true
STEP 6: Check the frontend of the default language, it is working
STEP 7: Change language. Meta fields are not showing.

And here is the big issue with WPML and not Crocoblock.
If you try to edit the second language (not with WPML tranlsation but as editing a post) you will notice that the metafields are properly displayed in the correct language but they are unchecked. This is what causes the whole issue because WPML cannot copy the "true" or "false" status for each meta field so we have to manually recheck those checkboxes or radios in the translated post too.

Lastly, another issue that I have discovered is that Jet Metafields (checkboxes and radios) cannot be displayed on the frontend as their labels but only as their values that are stored in the database. This is way too "buggy" because, who needs to show the database fileds to everyone? Last time I discussed this with the Crocoblock team they said that checkboxes and radios are supposed to be used with Jet Smart Filters, Well yes, we need them to be filtered but we also need to show to the customers the features of a product that we have already checked as checkboxes. 

So, for now the only working solution is using ACF checkboxes and radios that indeed can be translated and copied between translations as already prechecked so we don't need to recheck them in every language.

I hope I helped a little more, let's hope both plugins will manage to fix that soon.

**Comment by incircolo** [comment]
Hi 
Yes, I investigated the situation further and found a similar way to make it work (some problems I had were caused by the WAF blocking a POST request).
The main problem, however, that I noticed is that to be able to translate everything easily I have to use the WPML CTE, but in order to translate the metafields (checkbox) I need to disable CTE and translate via Wordpress. In this way, however, it becomes impossible to return to CTE, otherwise the associations between the checkbox and the new language would be lost.

Can ACF coexist properly with JetEngine? At this point I consider whether to manage these meta with ACF... or ... taxonomy could be used instead of metafields to define these features.

PS.
The workaround i have found now is to set "not translatable" the checkbox metafields, so the translated ones will keep shifting between CTE and Wordpress standard editor... but i'm not sure it's a convenient choice (they appears "translated" as we are translating admin labels with String Translation... we are creating new items in checkbox fields...). If we set "copy" or "translate" we are only make translatable the boolean value of checkbox fields, but ... as these translations are not working, we can set this to not translate and translate only the admin label... so we will going to add new items and not translating them... but at least it works (only if you have not too many post i think)

**Comment by torsten3d** [comment]
+1  Sad that it isn't fixed thee years after reporting.
